### PR TITLE
Add toolbar search field for nodes, small fixes when "refreshing"

### PR DESCRIPTION
**Pull request by hybridherbst**
Adds a little toolbar search field:
image

Search selects and frames nodes with titles containing the search string for now.
Possible future improvements would be tapping into the search provider etc, but already feels more manageable in complex graphs.

### CODE REVIEW

**Comment by Unity-Harry** [comment]
Just to let you know I've seen this, just been on holiday and strapped for time.
Hope to give it a proper look soon! :)

File: AssetDependencyGraph/Editor/AssetDependencyGraph.cs [hunk 1]
```diff
@@ -87,6 +88,25 @@ public void OnEnable()
...
+        ts.RegisterValueChangedCallback(x => {
+            if (string.IsNullOrEmpty(x.newValue)) {
+                m_GraphView.FrameAll();
+                return;
+            }
+            
+            m_GraphView.ClearSelection();
+            // m_GraphView.graphElements.ForEach(y => { // BROKEN, Case 1268337
+            m_GraphView.graphElements.ToList().ForEach(y => {
+                if (y is Node node && y.title.Contains(x.newValue)) {
```
Comment (Unity-Harry) [inline AssetDependencyGraph/Editor/AssetDependencyGraph.cs]:
```suggestion
                if (y is Node node && y.title.IndexOf(x.newValue, StringComparison.OrdinalIgnoreCase) >= 0) {
```
Not sure if it'd be worth us making this case insensitive, perhaps it's better to be specific when you're dealing with a large graph?

**Review by Unity-Harry — APPROVED** [review]
Great addition, thank you!
Just one small question about the search bar

File: AssetDependencyGraph/Editor/AssetDependencyGraph.cs (hunk 1 above)
Comment (hybridherbst) [reply AssetDependencyGraph/Editor/AssetDependencyGraph.cs]: I think let's go with insensitive for now; at some future point Searcher package integration would be cool but not for this PR :)

**Pull request merged**